Source/Core/DolphinQt/SkylanderPortal/SkylanderModifyDialog.cpp

```diff
@@ -0,0 +1,317 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "SkylanderModifyDialog.h"
+
+#include <Core/IOS/USB/Emulated/Skylanders/Skylander.h>
+#include <Core/System.h>
+
+#include <DolphinQt/QtUtils/SetWindowDecorations.h>
+
+SkylanderModifyDialog::SkylanderModifyDialog(QWidget* parent, u8 slot) : QDialog(parent)
+{
+  this->slot = slot;
+
+  layout = new QVBoxLayout;
+
+  IOS::HLE::USB::Skylander* skylander =
+      Core::System::GetInstance().GetSkylanderPortal().GetSkylander(slot);
+
+  figure = skylander->figure;
+  figureData = figure->GetData();
+
+  auto* hbox_name = new QHBoxLayout;
+  std::string name = "";
```

Comment:
This might be better as a `QString` to avoid the conversion on line 29 and  45 and then again on line 52.  You'd only need to convert on line 40.